src/ConversationList/ConversationItemModel.vala

```diff
@@ -190,4 +189,14 @@ public class Mail.ConversationItemModel : GLib.Object {
 
         return time;
     }
+
+    private static bool get_all_unread (Camel.FolderThreadNode? node) {
```

Comment:
PS: We should also return as soon as possible to avoid looping over all child nodes. So if the flag was already found:

```vala
if (!has_flag) {
    for (unowned ...) {
        has_flag = ...
        if (has_flag) {
            break;
```